sys/drv/evdev.c

```diff
@@ -469,10 +475,9 @@ static int evdev_open(vnode_t *v, int mode, file_t *fp) {
   WITH_MTX_LOCK (&evdev->ev_lock)
     LIST_INSERT_HEAD(&evdev->ev_clients, client, ec_link);
 
-  fp->f_ops = &evdev_fileops;
-  fp->f_data = client;
+  fp->f_data = dev;
 
-  return error;
+  return 0;
```

Comment:
why there isn't any kind of error returned?

Reply:
In the current implementation actually nothing can fail, including memory allocation due to `M_WAITOK` flag.